caesar_cipher.rb

```diff
@@ -0,0 +1,15 @@
+
+LETTERS_ARRAY = ("a".."z").to_a
+
+def caesar_cipher(string, shift_num)
+    chars_array = string.chars
+    ciphered_array = chars_array.map  do |char|
+        if LETTERS_ARRAY.index(char.downcase)
+            ciphered_index = LETTERS_ARRAY.index(char.downcase) + shift_num
+            ciphered_lowercase_char = LETTERS_ARRAY[ciphered_index] || LETTERS_ARRAY[ciphered_index - LETTERS_ARRAY.length]
+            char == char.upcase ? ciphered_lowercase_char.upcase : ciphered_lowercase_char
+        else char
+        end
```

Comment:
It's always best to "map out" with guards the problematic cases.
this is done in rails:
if it's inside function - use return
if it's inside block - use next (it's the equivelnt of js continue - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/continue)

```suggestion
        letter_index = LETTERS_ARRAY.index(char.downcase)
        next(char) if letter_index
        
        ciphered_index = LETTERS_ARRAY.index(char.downcase) + shift_num
        ciphered_lowercase_char = LETTERS_ARRAY[ciphered_index] || LETTERS_ARRAY[ciphered_index - LETTERS_ARRAY.length]
        char == char.upcase ? ciphered_lowercase_char.upcase : ciphered_lowercase_char
```

Reply:
Implemented but with refactor of extracting method that means I should use `return`
And also `unless` instead of `if` since it's a negative condition